src/backend/joanie/core/models/products.py

```diff
@@ -0,0 +1,253 @@
+            # this issue could be due to a bad setting or a bad resource_link filled,
+            # so we need to log this error to fix it quickly to joanie side
+            if lms is None:
+                order.state = enums.ORDER_STATE_FAILED
+                order.save()
+                enrollment.state = enums.ENROLLMENT_STATE_FAILED
+                enrollment.save()
+                logger.error(
+                    "No LMS configuration found for resource link: %s", resource_link
+                )
+            # now enroll user to lms course run and change joanie enrollment state to 'in_progress'
+            else:
+                lms_enrollment = lms.set_enrollment(user.username, resource_link)
+                if lms_enrollment["is_active"]:
+                    enrollment.state = enums.ENROLLMENT_STATE_IN_PROGRESS
+                    enrollment.save()
+        return order
+
+
+class ProductCourseRunPosition(models.Model):
+    """
+    ProductCourseRunPosition model allows to define order of each course runs to follow
+    for a product.
+    Some course runs could have the same position for a product (various sessions for example).
+    Validation of only one of them will be enough to pass to the next course run or certification.
+    """
+
+    product = models.ForeignKey(
+        Product,
+        verbose_name=_("product"),
+        related_name="course_runs_positions",
+        on_delete=models.CASCADE,
+    )
+    position = models.PositiveSmallIntegerField(_("position in product"))
+    course_run = models.ForeignKey(
+        courses_models.CourseRun,
+        verbose_name=_("course run"),
+        related_name="positions",
+        on_delete=models.RESTRICT,
+    )
+
+    class Meta:
+        db_table = "joanie_product_course_run_position"
+        verbose_name = _("Position of course runs in products")
+        verbose_name_plural = _("Positions of course runs in products")
+        unique_together = ("product", "course_run", "position")
+
+    def __str__(self):
+        return f"{self.product}: {self.position}/ {self.course_run}]"
+
+
+class Order(models.Model):
+    """
+    Order model represents and records details user's order (for free or not) to a course product
+    All course runs to enroll selected are defined here.
+    """
+
+    uid = models.UUIDField(default=uuid.uuid4, unique=True)
+    product = models.ForeignKey(
+        Product,
+        verbose_name=_("product"),
+        related_name="orders",
+        on_delete=models.RESTRICT,
+    )
+    course_runs = models.ManyToManyField(
+        courses_models.CourseRun, verbose_name=_("course runs")
+    )
+    owner = models.ForeignKey(
+        customers_models.User,
+        verbose_name=_("owner"),
+        related_name="orders",
+        on_delete=models.RESTRICT,
+    )
+    created_on = models.DateTimeField(_("created on"), default=timezone.now)
+    updated_at = models.DateTimeField(auto_now=True)
```

Comment:
I think it should be the same preposition as for "created on":
```suggestion
    updated_on = models.DateTimeField(auto_now=True)
```